### ISSUE TITLE
AbstractQuery::execute false positive

### ISSUE THREAD

**Issue by soullivaneuh** [phpstan/phpstan-doctrine#182]
# Sumarry

Given this code example:

```php
declare(strict_types=1);

namespace App\Repository;

use App\Entity\IpV4;
use Doctrine\ORM\EntityRepository;
use Doctrine\ORM\QueryBuilder;

/**
 * @extends EntityRepository<IpV4>
 */
class IpV4Repository extends EntityRepository
{
    public function findByMagic(): QueryBuilder
    {
        $qb = $this->createQueryBuilder('i');
        $qb->where('i.server = :server');

        return $qb->getQuery()->execute(['server' => 'test']);
    }
}
```

I have:

```
 ------ ------------------------------------------------------------------------------------------- 
  Line   Repository/IpV4Repository.php                                                              
 ------ ------------------------------------------------------------------------------------------- 
  21     Parameter #1 $parameters of method Doctrine\ORM\AbstractQuery::execute() expects           
         (Doctrine\Common\Collections\ArrayCollection&iterable)|null, array<string, string> given.  
 ------ ------------------------------------------------------------------------------------------- 
```

Not sure if it's a phpstan or doctrine issue.

### PHPStan version and packages

```
> composer bin phpstan show
phpstan/phpstan                  0.12.83 PHPStan - PHP Static Analysis Tool
phpstan/phpstan-doctrine         0.12.33 Doctrine extensions for PHPStan
phpstan/phpstan-phpunit          0.12.18 PHPUnit extensions and rules for PHPStan
phpstan/phpstan-strict-rules     0.12.9  Extra strict and opinionated rules for PHPStan
phpstan/phpstan-symfony          0.12.25 Symfony Framework extensions and rules for PHPStan
phpstan/phpstan-webmozart-assert 0.12.12 PHPStan webmozart/assert extension
```

### Doctrine related installed package

```
doctrine/annotations                     1.12.1            
doctrine/cache                           1.10.2            
doctrine/collections                     1.6.7             
doctrine/common                          3.1.2             
doctrine/data-fixtures                   1.5.0             
doctrine/dbal                            2.13.0            
doctrine/deprecations                    v0.5.3            
doctrine/doctrine-bundle                 2.3.1             
doctrine/doctrine-cache-bundle           1.4.0             
doctrine/doctrine-fixtures-bundle        3.4.0             
doctrine/doctrine-migrations-bundle      3.1.1             
doctrine/event-manager                   1.1.1             
doctrine/inflector                       1.4.3             
doctrine/instantiator                    1.4.0             
doctrine/lexer                           1.2.1             
doctrine/migrations                      3.1.1             
doctrine/orm                             2.8.4             
doctrine/persistence                     2.1.0             
doctrine/sql-formatter                   1.1.1             
```

**Comment by ondrejmirtes** [comment]
The problem is that PHPStan considers `ArrayCollection|mixed[]` to be `ArrayCollection of mixed`, not `array of mixed OR ArrayCollection of mixed`. This is expected behaviour. Doctrine needs to fix their typehints, for example to `ArrayCollection|array|mixed[]`.

**Comment by soullivaneuh** [comment]
Thanks for the explanation, I opened an issue accordingly.
